File: src/apps/mobile/ios/OpenBitFun/Infrastructure/StartupRevealPreference.swift
import Foundation

/// Installation-local presentation state, independent of accounts and app versions.
@MainActor
enum StartupRevealPreference {
    static func claim(defaults: UserDefaults = .standard) -> Bool {
        let key = "startup.brandRevealShown"
        guard !defaults.bool(forKey: key) else { return false }
        // Claim before playback so interruptions do not replay the reveal next launch.
        defaults.set(true, forKey: key)
        return true
    }
}

File: src/apps/mobile/android/app/src/main/kotlin/com/openbitfun/mobile/app/platform/StartupRevealPreference.kt
package com.openbitfun.mobile.app.platform

import android.content.Context

/** Installation-local presentation state; account changes and upgrades do not reset it. */
internal object StartupRevealPreference {
    @Synchronized
    fun claim(context: Context): Boolean {
        val preferences = context.getSharedPreferences("startup_presentation", Context.MODE_PRIVATE)
        if (preferences.getBoolean("brand_reveal_shown", false)) return false
        // Claim before playback, including when the first reveal is interrupted.
        preferences.edit().putBoolean("brand_reveal_shown", true).apply()
        return true
    }
}
